### DISCUSSION TITLE
Bookmarks suggestions for improvement

### DISCUSSION THREAD

**Discussion by Uzver123** [WerWolv/ImHex#1590] [Feedback]
1. Currently when creating bookmark that overlaps another bookmark the one that is coming first shows and the second one is not visible any any way. 
![image](https://github.com/WerWolv/ImHex/assets/143356096/c6e530ad-e6f7-4ba4-b435-dda0ce014273)
Could you make two bookmarks highlighting show at same time in editor view. 
Logic could be smaller range (red on screenshot) should show over the bigger one (green) you could check by the range beginning and end.

2. Another option is allow to set Opacity 0-255 for bookmark so green bookmark can be half way transparent then red one will show.
Right now you have opacity control built into color selection, however when selecting opacity that way it does not reveal bookmark that is underneath rather showing empty space.
This example Blue bookmark overlay over Red and Green bookmarks (its not adjacent to them)
![3](https://github.com/WerWolv/ImHex/assets/143356096/57ccc37f-0986-4241-8105-1ee9e1f044f3)

3. Allow groups for bookmarks in case someone loads 1000 bookmarks its best to have them organized in collapsible/expandable groups than have list of 1000 records.
Right now i managed this workaround, but its not ideal.
![image](https://github.com/WerWolv/ImHex/assets/143356096/1e57dcd6-9643-498a-9f34-0b49c3621974)

4. Bookmarks when exported use byte offset, yet in editor they use only hex offset, please allow to enter either hex or decimal (byte) offset in editor. 
![image](https://github.com/WerWolv/ImHex/assets/143356096/bcdc2a2f-68e0-4001-9a5a-3feed08a95fa)

5. Allow to see value of bookmark right inside bookmarks tab also ability to select data type for bookmark e.g. float/double/decimal etc.
Example 
![image](https://github.com/WerWolv/ImHex/assets/143356096/f820575b-9883-4b9c-bcdb-81f2fb859646)

6. Allow bookmark visibility to be turned on/off when off it does not show highlighting in the Hex Editor view panel.

7. Allow to resize existing bookmark visually inside the Hex Editor by dragging its beginning/end edge.
![1](https://github.com/WerWolv/ImHex/assets/143356096/0a945550-8853-495b-a93e-c969a001d8e1)

8. same as issue 2 (described above) bookmarks are currently hiding the `pattern `if they overlay over it, please allow to work in tandem or allow to select what will have higher priority `bookmarks `or `patterns`.
![image](https://github.com/WerWolv/ImHex/assets/143356096/6a94c936-5aab-4e9d-8e00-d9c6a8c04be8)

9. Change bookmarks order, currently when bookmarks can only be dragged over another bookmark and this will cause them to swap spaces, please introduce way to change order rather than replace one bookmark with another.
![image](https://github.com/WerWolv/ImHex/assets/143356096/4df08b41-8253-424f-b357-a20a44bff121)

**Comment by WerWolv** [comment]
Hey, sorry for not responding earlier. I believe we made quite some progress with many of these features already and I added some more that you mentioned just now.

- 1/2/8) Any highlights in the hex editor view may now overlap each other. Their color gets blended together so all of them stay visible at all times.
- 5) This can all be done using the pattern language and is much better suited for it. It lets you highlight and decode data just the same as you're showing with 010 Editor there.
- 6) I just implemented a toggle now to show/hide the highlighting of any bookmark
- 9) Bookmarks can now be dragged around and inserted between other bookmarks without switching place

Hope this helps!
